### DISCUSSION TITLE
How to disable clickable row ?

### DISCUSSION THREAD

**Discussion by xBigDaddyx** [filamentphp/filament#6189] [Help] [answered]
i tried 

protected function getTableRecordUrlUsing(): ?Closure
{
    return null;
}

on my resource, seems not working.

**Comment by rmitesh** [comment, marked as answer]
Hi @Terefai 

Instead of adding in the main resource, you have to add in your List class.

`PlanResource > Pages > ListPlans`

```php
use Closure;

class ListPlans extends ListRecords
{
    // ...
    
    protected function getTableRecordUrlUsing(): ?Closure
    {
        return null;
    }
}
```

**Reply by xBigDaddyx** [reply]
Ahh i see, thanks.

**Reply by davodsaraei** [reply]
@rmitesh 
does not work!

**Reply by magarrent** [reply]
Thanks! Be careful, on the Documentation->Tables it shows:

`getTableRecordActionUsing()`

I'm using Admin panel and you must use **getTableRecordUrlUsing** instead of _getTableRecordActionUsing_

**Comment by davodsaraei** [comment]
The `getTableRecordUrlUsing` method is deprecated and must override the `table` method. So the following code should be worked.

```PlanResource > Pages > ListPlans```
```php
protected function makeTable(): Table
{
    return parent::makeTable()->recordUrl(null);
}
```

**Reply by danharrin** [reply]
You don't need to override `makeTable()`, just chain `->recordUrl()` onto `$table` in `table()`

**Reply by anselmobattisti** [reply]
It worked like a charm!

**Reply by xiaomlove** [reply]
> You don't need to override `makeTable()`, just chain `->recordUrl()` onto `$table` in `table()`

is this only works in V3?

**Reply by solstyc** [reply]
> The `getTableRecordUrlUsing` method is deprecated and must override the `table` method. So the following code should be worked.
> 
> `PlanResource > Pages > ListPlans`
> 
> ```
> protected function makeTable(): Table
> {
>     return parent::makeTable()->recordUrl(null);
> }
> ```

Why is it not in the upgrade guide as a breaking change that getTableRecordUrlUsing no longer works? I was getting errors and the only thing that helped me was Grok

**Comment by untillnesss** [comment]
it still can click the row. How do I disable it? so it can't be clicked.

**Reply by danharrin** [reply]
What happens when you click

**Reply by leandrocfe** [reply]
`->recordAction(null)` ?
https://github.com/filamentphp/filament/discussions/8142#discussioncomment-6846440

**Reply by untillnesss** [reply]
@danharrin when clicked it will lead to the edit page. or if there is a viewing page, he will go there. even though I didn't expect that.

**Reply by leandrocfe** [reply]
did you try `$table->recordAction(null)` ?

**Reply by danharrin** [reply]
`->recordUrl(null)` is definitely the solution for this, you must be putting it in the wrong place or something

**Reply by sinnbeck** [reply]
Using `->recordUrl(null)` will change it from edit to view if there is a ViewAction on the table. It does not disable it in that case

![click-to-view](https://github.com/filamentphp/filament/assets/13980140/73694e65-b3e7-4ccf-9d7a-3316afb11e1b)

**Comment by jdion84** [comment]
i've overridden `getTableRecordActionUsing` and `getTableRecordUrlUsing` methods to return `null` in the list page class and the rows are still clickable

not a huge deal, but would be cool if maybe at the table-level we could have a method like `disableClickableRows`

**Comment by WillieOng-HK** [comment]
XXXResource Class  > static function table > return $table

```
... 
->recordUrl(
    fn (Model $record) => null,
);
```

[see](https://filamentphp.com/docs/3.x/tables/advanced#record-urls-clickable-rows)

**Reply by danharrin** [reply]
Or just `recordUrl(null)`, and then `recordAction(null)` if it then starts to open a modal

**Reply by Integrityby-design** [reply]
I came searching for a 'Disable Clickable Rows' function for the Resource table. This option should be chained inside your Main Resource file. I wanted to disable the Edit page from loading when a row is clicked. I'm running v3.0.0 Filament.

->recordUrl(null)    // works

```
public static function table(Table $table): Table
   {
       return $table
             ->recordUrl(null)    // this should be enough
                 ...
```
I also tested `->recordAction(null)` and this doesn't seem to work. Though `->recordUrl(null)` did not pop up a modal for me.

**Reply by melodymoselle** [reply]
would be helpful to add this to the docs...., the `recordAction(null)` is what I was looking for.

**Comment by alisalehi1380** [comment]
```php
     public static function table(Table $table): Table
    {
        return $table
            ->recordAction(null)   // => works for me. ✅
            ->columns([])
            ->filters([])
            ->actions([
                EditAction::make()->closeModalByClickingAway(false),
            ])
            ->bulkActions([
                BulkActionGroup::make([
                    DeleteBulkAction::make(),
                ]),
            ]);
    } 
```

**Comment by pratheeshimmanuel** [comment]
->recordAction(null)   // => works well for me. ✅

**Comment by atiladelcanton** [comment]
This option not work for me :(

**Comment by rtraselbd** [comment]
Use it

->recordUrl(null)
->recordAction(null)

**Comment by jeffegiovani** [comment]
I tried to use in my _**Panel Provider**_ with `configureUsing()`, like this:

```
Filament\Tables\Table::configureUsing(function (Filament\Tables\Table $table): void {
            $table
            ->recordUrl(null)
            ->recordAction(null);
});
```

But without success, the rows can be clicked.

Maybe this methods can be included in `configureUsing()`, because add this in every tables separately is very expensive.

**Reply by dharen008** [reply]
Same issue it is not working on global settings, even i added the isImportant: true it doesn't work

Table::configureUsing( function ( Table $table ): void {
			$table
				->recordAction( null )
				->recordUrl( null );
		}, isImportant: true );
		
FYI @danharrin

**Reply by zsnakeee** [reply]
I've submitted a PR (#15350) that fixes this issue. The problem was that default values for `recordAction` and `recordUrl` were being applied even when these values were explicitly set to `null`.

The fix adds tracking of whether these values have been explicitly set by the user, ensuring that default values are only applied when the user hasn't set them. This allows proper nullification of record actions and URLs in both regular tables and relation managers.

I've also added comprehensive tests to verify the behavior and prevent regression. You can check out the PR here: https://github.com/filamentphp/filament/pull/15350

**Reply by Renatomelo88** [reply]
> Tentei usar no meu _**Panel Provider**_ com `configureUsing()`, assim:
> 
> ```
> Filament\Tables\Table::configureUsing(function (Filament\Tables\Table $table): void {
>             $table
>             ->recordUrl(null)
>             ->recordAction(null);
> });
> ```
> 
> Mas sem sucesso, as linhas podem ser clicadas.
> 
> Talvez esses métodos possam ser incluídos `configureUsing()`, porque adicioná-los a cada tabela separadamente é muito caro.


try

        Column::configureUsing(function (Column $section): void {
            $section
                ->disabledClick();
        });

**Reply by jeffegiovani** [reply]
> try
> 
> ```
>     Column::configureUsing(function (Column $section): void {
>         $section
>             ->disabledClick();
>     });
> ```

This works fine (suggested by @Renatomelo88 ), but pay attention with TextColumn->action() at V4.

Need to "reset" disableClick() method:

```
TextColumn::make('property_contacts_count')
    ->disabledClick(false)
    ->action(Action::make('property_contacts_preview')
        ->modalContent(
            fn($record) => view('modal-contacts', ['contacts' => $record->propertyContacts])
        )
    ->alignCenter()
```
